packages/snap/src/keyring.test.ts

```diff
@@ -43,20 +49,27 @@ jest.mock('../src/utils/ethers', () => ({
     new ethers.Wallet(privateKey, ethers.provider),
 }));
 
-jest.mock('uuid', () => ({
-  v4: () => mockAccountId,
-}));
+jest.mock('uuid', () => {
+  return {
+    v4: () => {
+      accountCreationCount += 1;
+      return accountCreationCount === 1
+        ? mockAccountId
+        : jest.requireActual('uuid').v4();
+    },
```

Comment:
We can just return a `jest.fn()` instead and mock return the the values that you want.

Reply:
From my research and efforts, I don't think this is possible without significantly modifying the way our tests is set up, so I think we can leave the implementation as is for now.
see [5f5e26a](https://github.com/MetaMask/snap-account-abstraction-keyring/pull/51/commits/5f5e26aec9dcd4a9b44b92c79f4afcea0e649a39)